src/test/java/core/basesyntax/service/impl/FruitShopServiceImplTest.java

```diff
@@ -0,0 +1,82 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitShopService;
+import core.basesyntax.strategy.handler.OperationHandler;
+import core.basesyntax.strategy.handler.impl.BalanceHandler;
+import core.basesyntax.strategy.handler.impl.OperationStrategyImpl;
+import core.basesyntax.strategy.handler.impl.PurchaseHandler;
+import core.basesyntax.strategy.handler.impl.ReturnHandler;
+import core.basesyntax.strategy.handler.impl.SupplyHandler;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class FruitShopServiceImplTest {
+    private static final String APPLE = "apple";
+    private static final String BANANA = "banana";
+    private static final String ORANGE = "orange";
+    private static final String PEAR = "pear";
+    private FruitShopService fruitShopService;
+
+    @BeforeEach
+      public void setUp() {
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap
+                = new HashMap<>();
+        operationHandlerMap.put(FruitTransaction.Operation.BALANCE, new BalanceHandler());
```

Comment:
You do not need to verify every operation of your strategy in tests for FruitShopService. 1 operation just to test that this service calls the operation correctly is enough.
Refer to common mistakes:
![image](https://github.com/mate-academy/jv-fruit-shop-tests/assets/82539553/5e81d626-92aa-41fa-b63f-37102c50f230)

Reply:
not resolved